chains/ethereum/connection.go

```diff
@@ -24,6 +25,17 @@ import (
 
 var _ chains.Connection = &Connection{}
 
+var NonceLock = &sync.Mutex{}
```

Comment:
This should be parameter on Connection, no need to be a pointer

Reply:
By parameter, do you mean we should be passing in a Mutex lock struct when we call newConnection, or just have it as a field in connection that we create with the connection object?

Reply:
It should be a field

Reply:
Ok, that's how I implemented in the last push, thanks for clarification.

Reply:
Still exists

Reply:
sorry